### PR TITLE
Hw 7

### PR DESCRIPTION
**Pull request by povilassta**
Added responsive navbar, main window with search function, login window was added in previous pr

### CODE REVIEW

File: client/src/app/home/home/account-card/account-card.component.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<mat-card
+  (click)="onCardClick()"
+  [ngClass]="{ selected: account._id === accountService.activeAccount?._id }"
+  class="card"
+>
+  <mat-card-subtitle class="card__title">{{ account.title }}</mat-card-subtitle>
+  <mat-card-title class="card__amount"
+    >{{ 123 }} {{ account.currency.code }}</mat-card-title
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/account-card/account-card.component.html:8]: 123? mock data?

Comment (povilassta) [reply client/src/app/home/home/account-card/account-card.component.html:8]: I had some problems implementing, the sum calculation in backend so in order to move on I used mock data here. The plan is to implement the login on backend today.

File: client/src/app/home/home/transaction-card/services/transaction.service.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import { Observable, tap } from 'rxjs';
+import { Transaction } from '../transaction.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TransactionService {
+  public transactions!: Transaction[];
+  public filteredTransactions!: Transaction[];
+  public isIncomeFilter: boolean = false;
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/transaction-card/services/transaction.service.ts:12]: you don't need isIncomeFilter and isExpenseFilter, you can leave one of them

File: client/src/app/home/home/transaction-card/services/transaction.service.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          next: (res: any) => {
+            this.transactions = res;
+            this.filteredTransactions = res;
+          },
+        })
+      );
+  }
+
+  public filterTransactions(type?: string): void {
+    if (!type) {
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/transaction-card/services/transaction.service.ts:31]: do you really need to store filtered transactions? This function can return just return it

File: client/src/app/home/home/transaction-card/transaction-card.component.sass [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    justify-content: space-between
+    background: white
+    margin-bottom: 0.4rem
+    padding: 0.5rem
+    border-radius: 15px
+    text-align: center
+    line-height: 2rem
+    &__category
+        width: 5rem
+        background: #d3d3d3
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/transaction-card/transaction-card.component.sass:14]: Do you have a file with color-constants? It should be in this file

File: client/src/app/home/home/transaction-card/transaction-card.component.html [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      income: transaction.type === 'income',
+      expense: transaction.type === 'expense'
+    }"
+    class="transaction-icon"
+  >
+    {{ transaction.type === 'income' ? 'arrow_downward' : 'arrow_upward' }}
+  </mat-icon>
+  <div class="card__category">{{ transaction.category.name }}</div>
+  <div class="card__title">{{ transaction.title }}</div>
+  <div class="card__date">{{ formatDate(transaction.transactionDate) }}</div>
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/transaction-card/transaction-card.component.html:13]: use pipe for formatting data, remove function

File: client/src/app/home/home/transaction-card/transaction-card.component.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  selector: 'app-transaction-card',
+  templateUrl: './transaction-card.component.html',
+  styleUrls: ['./transaction-card.component.sass'],
+})
+export class TransactionCardComponent implements OnInit {
+  @Input() transaction!: Transaction;
+
+  constructor() {}
+
+  public formatDate(date: Date) {
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/transaction-card/transaction-card.component.ts:15]: remove

File: client/src/app/home/home/transaction-card/transaction.model.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+export interface Transaction {
+  _id: string;
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/transaction-card/transaction.model.ts:2]: Move all interfaces in folder models

File: client/src/app/home/home/transaction-card/services/transaction.service.ts (hunk 3 above)
Comment (povilassta) [reply client/src/app/home/home/transaction-card/services/transaction.service.ts:31]: It doesnt update the ui if it's not tied to a variable. But perhaps i just lack some knowledge of how to do it without the variable.

File: client/src/app/home/home/account-card/account-card.component.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  templateUrl: './account-card.component.html',
+  styleUrls: ['./account-card.component.sass'],
+})
+export class AccountCardComponent implements OnInit {
+  @Input() public account!: Account;
+
+  public onCardClick(): void {
+    this.accountService.activateAccount(this.account._id);
+    this.transactionService.getTransactions(this.account._id).subscribe({
+      next: (data) => (this.transactionService.transactions = data),
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/account-card/account-card.component.ts:17]: oh, i missed it.. It is bad to change value of service from component. If you need transactions here, create variable in this component

File: client/src/app/home/home/home.component.html [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  </div>
+
+  <div class="main__options">
+    <div class="filter-buttons">
+      <button
+        [ngClass]="{ selected: transactionService.isIncomeFilter }"
+        class="filter-buttons__income"
+        mat-mini-fab
+        aria-label="income"
+        (click)="transactionService.filterTransactions('income')"
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/home.component.html:52]: create function in your component, this function will call transactionService.filterTransactions('income').
transactionService.filterTransactions('income') should return array and you will save it in variable filteredTransactions in this component. Something like this

File: client/src/app/home/home/home.component.html [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+      >
+        <mat-icon>arrow_downward</mat-icon>
+      </button>
+      <button
+        [ngClass]="{ selected: transactionService.isExpenseFilter }"
+        class="filter-buttons__expense"
+        mat-mini-fab
+        aria-label="expense"
+        (click)="transactionService.filterTransactions('expense')"
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/home.component.html:61]: same for this

File: client/src/app/home/home/account-card/account-card.component.ts (hunk 8 above)
Comment (povilassta) [reply client/src/app/home/home/account-card/account-card.component.ts]: oh yeah, this does nothing, I forgot to remove it.

File: client/src/app/home/home/home.component.ts [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    public transactionService: TransactionService,
+    private communicationService: CommunicationService
+  ) {
+    this.communicationService.componentMethodCalled$.subscribe(() => {
+      this.filterTransactions();
+    });
+  }
+
+  public ngOnInit(): void {
+    this.accountsService.getAccounts().subscribe({
```
Comment (AnastasiaHarost) [inline client/src/app/home/home/home.component.ts:38]: add pipe(UntilDestroyed(this))  -- to destroy observable

Comment (AnastasiaHarost) [reply client/src/app/home/home/home.component.ts:38]: need to add this for every component when you subscribe on observable

File: client/src/app/models/transaction.model.ts [hunk 12]
```diff
@@ -0,0 +1,14 @@
+export interface Transaction {
+  _id: string;
+  title: string;
+  type: string;
+  transactionDate: Date;
+  accountId: string;
+  amount: Number;
+  category: {
```
Comment (AnastasiaHarost) [inline client/src/app/models/transaction.model.ts:8]: create separate interface for currency and reuse it here

**Review by AnastasiaHarost — APPROVED** [review]
good)

**Pull request merged**